### DISCUSSION TITLE
UseContractRead returns indefine but when we switch the network it works fine

### DISCUSSION THREAD

**Discussion by Anup9444** [wevm/wagmi#1689] [Q&A]
`
  const { data, error, isLoading } = useContractRead({
    address: token.address,
    abi: ABI.ERC20ABI.allowance,
    functionName: "allowance",
    args: [connectedAddress, chainData.instantContract],
    // cacheTime: WAGMI_READ_CACHE_TIMEOUT_MS,
    enabled: !!address && !token.isNative,
  });
`
when i call this hook for first time it returne `undefine` and error message with
 `react_devtools_backend.js:4012 Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="allowance(address,address)", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.7.0)`
 and after when i switch the current network(chain)  to another network(chain) then it works properly

i console the argument and address and every things are correct ...
i am facing this for long time .....pleases help me out 

i used `    "wagmi": "^0.8.7",`

**Comment by SaadNiaziDev** [comment]
Contract chain and your connected chain must be same to commuicate. Are you using signer | | provider for contract?

**Reply by Anup9444** [reply]
i am provider .....
when i console log the provider it was on chain 137 although i was on goerli network . and after i change my network to any other network and come back to goerli again now their isn't any error and reads data from blockchain properly

**Comment by SaadNiaziDev** [comment]
1-Clear localstorage/unlink account with your web
2- import your chain from wagmi/chains e.g
------------import { avalanche, bsc, mainnet } from 'wagmi/chains'
3- use wagmi useSigner hook  e.g
------------const { data: signer } = useSigner({
--------------chainId: bsc.id,                   (chain on which contract is deployed)
------------});
4- use wagmi useContract hook e.g 
------------const contract = useContract({
---------------address: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
---------------abi: ensRegistryABI,
---------------signerOrProvider: signer ,
------------ })

**Comment by Anup9444** [comment]
sorry, i didn't get it
i was getting this error while reading the allowance from useContractRead hook,my code look like this 
` 
 const { chainId } = useChain();
  const { address } = useAccount(); 
  const connectedAddress = address ?? `0x`;
  const chainData = Assets[chainId];
  const token = chainData.tokens[tokenSymbol];
  const { data, error, isLoading } = useContractRead({
    address: token.address,
    abi: ABI.ERC20ABI.allowance,
    functionName: "allowance",
    args: [connectedAddress, chainData.instantContract],
  }); 
`

**Comment by SaadNiaziDev** [comment]
Could u provide a recording for better understanding?
